tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspClientTool/TspClientTool.cs

```diff
@@ -0,0 +1,476 @@
+                args.Add(value);
+            }
+        }
+
+        private Task<(bool Success, string? ErrorMessage, int Duration)> ExecuteTspClientCommand(List<string> args)
+        {
+            var startTime = DateTime.UtcNow;
+            
+            try
+            {
+                logger.LogInformation("Executing npx @azure-tools/typespec-client-generator-cli with args: {Args}", string.Join(" ", args));
+
+                var arguments = string.Join(" ", args.Select(arg => arg.Contains(" ") ? $"\"{arg}\"" : arg));
+                
+                string output;
+                var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+                if (isWindows)
+                {
+                    output = RunProcess("cmd.exe", $"/C npx @azure-tools/typespec-client-generator-cli {arguments}", Environment.CurrentDirectory);
+                }
+                else
+                {
+                    output = RunProcess("npx", $"@azure-tools/typespec-client-generator-cli {arguments}", Environment.CurrentDirectory);
+                }
+
+                var duration = (int)(DateTime.UtcNow - startTime).TotalMilliseconds;
+                var success = !output.Contains("failed") && !output.Contains("error");
+
+                if (!success)
+                {
+                    SetFailure(1);
+                    var errorMessage = $"tsp-client command failed: {output}";
+                    
+                    // Add helpful context to common errors
+                    if (errorMessage.Contains("MODULE_NOT_FOUND") || errorMessage.Contains("command not found"))
+                    {
+                        errorMessage += "\n\nSuggestion: Make sure Node.js and npm are installed, and @azure-tools/typespec-client-generator-cli is available via npx.";
+                    }
+                    else if (errorMessage.Contains("tspconfig.yaml"))
+                    {
+                        errorMessage += "\n\nSuggestion: Ensure the tspconfig.yaml file exists and is properly formatted.";
+                    }
+                    
+                    return Task.FromResult((false, (string?)errorMessage, duration));
+                }
+
+                logger.LogInformation("npx @azure-tools/typespec-client-generator-cli completed successfully");
+                return Task.FromResult((true, (string?)null, duration));
+            }
+            catch (Exception ex)
+            {
+                var exceptionDuration = (int)(DateTime.UtcNow - startTime).TotalMilliseconds;
+                logger.LogError(ex, "Failed to execute npx @azure-tools/typespec-client-generator-cli command");
+                SetFailure(1);
+                return Task.FromResult((false, (string?)ex.Message, exceptionDuration));
+            }
+        }
+
+        private static string RunProcess(string command, string args, string workingDirectory)
+        {
+            var processInfo = new ProcessStartInfo
```

Comment:
@benbp I think we should add a process running helper to avoid this process running step in each tool. I will add a helper class.